test/common/common/utility_test.cc

```diff
@@ -116,6 +117,54 @@ TEST(DateUtil, NowToMilliseconds) {
   EXPECT_EQ(12345067, DateUtil::nowToMilliseconds(test_time));
 }
 
+TEST(OutputBufferStream, All) {
+  // Fails on write to empty buffer
+  {
+    std::string data = "123";
+    OutputBufferStream ostream{nullptr, 0};
+    ASSERT_TRUE(ostream.good());
+
+    ostream << data;
+
+    EXPECT_TRUE(ostream.bad());
+  }
+
+  // Can write to buffer
+  {
+    std::string data = "123";
+    std::string buffer(3, 'x');
+
+    OutputBufferStream ostream{buffer.data(), buffer.size()};
+    ostream << data << std::endl;
```

Comment:
Note that this test actually does write more to the buffer than it can hold since `std::endl` expands to some combination of '\r' and '\n'.